packages/ember-intl/addon/services/intl.ts

```diff
@@ -0,0 +1,397 @@
+      return '';
+    }
+
+    const intlShape = this.getIntlShape(options?.locale);
+
+    const descriptor =
+      typeof value === 'object'
+        ? value
+        : {
+            defaultMessage: value,
+            description: undefined,
+            id: value,
+          };
+
+    if (options?.htmlSafe) {
+      const output = formatMessage(
+        intlShape,
+        descriptor,
+        escapeFormatMessageOptions(options),
+      );
+
+      return htmlSafe(output) as unknown as string;
+    }
+
+    return formatMessage(intlShape, descriptor, options);
+  }
+
+  formatNumber(
+    value: FormatNumberParameters[0] | undefined | null,
+    options?: FormatNumberParameters[1] & {
+      locale?: string;
+    },
+  ): string {
+    if (value === undefined || value === null) {
+      return '';
+    }
+
+    const intlShape = this.getIntlShape(options?.locale);
+
+    return formatNumber(intlShape, value, options);
+  }
+
+  formatRelative(
+    value: FormatRelativeParameters[0] | undefined | null,
+    options?: FormatRelativeParameters[2] & {
+      locale?: string;
+      unit?: FormatRelativeParameters[1];
+    },
+  ): string {
+    if (value === undefined || value === null) {
+      return '';
+    }
+
+    const intlShape = this.getIntlShape(options?.locale);
+
+    return formatRelative(intlShape, value, options?.unit, options);
+  }
+
+  formatTime(
+    value: FormatTimeParameters[0] | undefined | null,
+    options?: FormatTimeParameters[1] & {
+      locale?: string;
+    },
+  ): string {
+    if (value === undefined || value === null) {
+      return '';
+    }
+
+    const intlShape = this.getIntlShape(options?.locale);
+
+    return formatTime(intlShape, value, options);
+  }
+
+  setLocale(locale: string | string[]): void {
+    const proposedLocale = convertToArray(locale);
+
+    if (hasLocaleChanged(proposedLocale, this._locale)) {
+      this._locale = proposedLocale;
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Argument of type 'Timer | undefined' is not assignable to parameter of type 'EmberRunTimer | undefined'
+      // eslint-disable-next-line ember/no-runloop
+      cancel(this._timer);
+
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Argument of type 'Timer | undefined' is not assignable to parameter of type 'EmberRunTimer | undefined'
+      // eslint-disable-next-line ember/no-runloop
+      this._timer = next(() => {
+        this._eventEmitter.emit('localeChanged');
+
+        this.updateDocumentLanguage();
+      });
+    }
+
+    this.updateIntl(locale);
+  }
+
+  t(
+    key: string,
+    options?: FormatMessageParameters[1] & {
+      htmlSafe?: boolean;
+      locale?: string;
+    },
+  ): string {
+    const locales = options?.locale ? [options.locale] : this._locale!;
+
+    let translation: string | undefined;
+
+    for (let i = 0; i < locales!.length; i++) {
```

Comment:
**breaking**

`options.default` is no longer allowed. End-developers are also responsible for ensuring that `key` is a string.